src/main/java/org/jenkinsci/plugins/parallel_test_executor/MultipleBinaryFileParameterFactory.java

```diff
@@ -0,0 +1,119 @@
+package org.jenkinsci.plugins.parallel_test_executor;
+
+import com.google.common.collect.Lists;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.AbstractBuild;
+import hudson.model.Action;
+import hudson.model.FileParameterValue;
+import hudson.model.ParametersAction;
+import hudson.model.TaskListener;
+import hudson.plugins.parameterizedtrigger.AbstractBuildParameterFactory;
+import hudson.plugins.parameterizedtrigger.AbstractBuildParameterFactoryDescriptor;
+import hudson.plugins.parameterizedtrigger.AbstractBuildParameters;
+import hudson.plugins.parameterizedtrigger.FileBuildParameterFactory;
+import hudson.util.IOException2;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Essentially a copy-paste of {@link hudson.plugins.parameterizedtrigger.BinaryFileParameterFactory} that takes a
+ * list of mappings "name -> filePattern" to generate parameters.
+ *
+ * @author Vincent Latombe <vincent@latombe.net>
+ */
+public class MultipleBinaryFileParameterFactory extends AbstractBuildParameterFactory {
+    public static class Tuple {
```

Comment:
Makes sense to set a more descriptive class name. Tuple may become a problem if you decide to add another field